packages/astro/src/core/errors/errors-data.ts

```diff
@@ -1260,6 +1260,23 @@ export const RewriteWithBodyUsed = {
 		'Astro.rewrite() cannot be used if the request body has already been read. If you need to read the body, first clone the request.',
 } satisfies ErrorData;
 
+/**
+ * @docs
+ * @description
+ * `Astro.rewrite()` can't be used to rewrite an on-demand route with a static route, when using the `"server"` output.
+ *
+ */
+export const ForbiddenRewrite = {
+	name: 'ForbiddenRewrite',
```

Comment:
Just noting that there may eventually be other *kinds* of "forbidden" rewrites, so not sure whether you want to add more context like `ForbiddenRewriteToStatic` or something like that?

Reply:
I'd prefer not. The kind of error would be the same, but we can customise the message based on the situation.